pkg/migration/controllers/migrationschedule.go

```diff
@@ -534,3 +572,47 @@ func (m *MigrationScheduleController) createCRD() error {
 	}
 	return apiextensions.Instance().ValidateCRDV1beta1(resource, validateCRDTimeout, validateCRDInterval)
 }
+
+// reconcileRemoteMigrationSchedule needs to send the updated hash value if there is a change in the source migration schedule
+func (m *MigrationScheduleController) reconcileRemoteMigrationSchedule(sourceMigrationSchedule *stork_api.MigrationSchedule, destMigrationSchedule *stork_api.MigrationSchedule) (uint64, error) {
+	var updatedHash uint64
+	// get the hash of the source migration schedule
+	sourceMigrationScheduleHash, err := getMigrationScheduleHash(sourceMigrationSchedule)
+	if err != nil {
+		return updatedHash, err
+	}
+
+	var destMigrationScheduleHash uint64
+	// check if destination migrationschedule has object has annotation and get the hash value
+	if destMigrationSchedule.Annotations == nil {
+		destMigrationSchedule.Annotations = make(map[string]string)
+	}
+	if val, ok := destMigrationSchedule.GetAnnotations()[resourcecollector.StorkResourceHash]; ok {
+		destMigrationScheduleHash, err = strconv.ParseUint(val, 10, 64)
+		if err != nil {
+			return updatedHash, fmt.Errorf("unable to parse hash value %v of destination migration schedule, err: %v", val, err)
+		}
+		if sourceMigrationScheduleHash != destMigrationScheduleHash {
+			log.MigrationScheduleLog(sourceMigrationSchedule).Infof("Source migration schedule hash %v, destination migration schedule hash %v", sourceMigrationScheduleHash, destMigrationScheduleHash)
+			updatedHash = sourceMigrationScheduleHash
+		}
+	} else {
+		// get the hash of the dest migration schedule
+		updatedHash, err = getMigrationScheduleHash(destMigrationSchedule)
```

Comment:
Or you could set the sourceHash here when you know that the annotation for hash on the destination side is empty.

Reply:
Right, this can be directly be the sourceHash. Will modify this.